File: core/src/main/java/com/example/core/CoreModule.java
package com.example.core;

import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

@Module
public class CoreModule {

    @Singleton
    @Provides
    MyCoreThing provideMyCoreThing() {
        return new MyCoreThing();
    }
}

File: core/src/main/java/com/example/core/CoreModule.kt
package com.example.core

import com.example.net.NetModule
import dagger.Module
import dagger.Provides

import javax.inject.Singleton

@Module
class CoreModule {

    @Singleton
    @Provides
    internal fun provideMyCoreThing(): MyCoreThing {
        return MyCoreThing()
    }
}
